test/crd.go

```diff
@@ -186,46 +186,17 @@ func ConfigurationWithBuild(namespace string, names ResourceNames, build *v1alph
 // LatestService returns a RunLatest Service object in namespace with the name names.Service
 // that uses the image specified by names.Image.
 func LatestService(namespace string, names ResourceNames, options *Options, fopt ...v1alpha1testing.ServiceOption) *v1alpha1.Service {
-	svc := &v1alpha1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      names.Service,
-		},
-		Spec: v1alpha1.ServiceSpec{
-			RunLatest: &v1alpha1.RunLatestType{
-				Configuration: *ConfigurationSpec(ImagePath(names.Image), options),
-			},
-		},
-	}
+	a := append([]v1alpha1testing.ServiceOption{v1alpha1testing.WithRunLatestConfigSpec(*ConfigurationSpec(ImagePath(names.Image), options))}, fopt...)
```

Comment:
I'd reverse the order:
```
fopt = append(fopt, v1alpha1testing.WithRunLatestConfigSpec(*ConfigurationSpec(ImagePath(names.Image), options)))
return ... , fopt...)
```

Reply:
The original logic (before the re-factoring) sets the service config spec  before applying any other ServiceOptions. (So, I am pre pending the `WithRunLatestConfigSpec` here)  Reversing the order (i.e. appending `WithRunLatestConfigSpec`) would change the original logic/intention.   What is worse is that we will get seg fault when subsequent  service options try to reference the undefined service config spec.

Reply:
I see, they might operate on the runLatest config.